ms2deepscore/MS2DeepScoreMonteCarlo.py

```diff
@@ -0,0 +1,171 @@
+        self.output_vector_dim = self.model.base.output_shape[1]
+        self.progress_bar = progress_bar
+        self.partial_model = self._create_monte_carlo_encoder()
+
+    def _create_input_vector(self, binned_spectrum: BinnedSpectrumType):
+        """Creates input vector for model.base based on binned peaks and intensities"""
+        X = np.zeros((1, self.input_vector_dim))
+
+        idx = np.array([int(x) for x in binned_spectrum.binned_peaks.keys()])
+        values = np.array(list(binned_spectrum.binned_peaks.values()))
+        X[0, idx] = values
+        return X
+
+    def _create_monte_carlo_encoder(self):
+        """Rebuild base network with training=True"""
+        dims = []
+        for layer in self.model.base.layers:
+            if "dense" in layer.name:
+                dims.append(layer.units)
+            if "dropout" in layer.name:
+                dropout_rate = layer.rate
+
+        # re-build encoder network with dropout layers always on
+        encoder = self.model.get_base_model(input_dim=self.input_vector_dim,
+                                            dims=dims,
+                                            embedding_dim=self.output_vector_dim,
+                                            dropout_rate=dropout_rate,
+                                            dropout_always_on=True)
+        encoder.set_weights(self.model.base.get_weights())
+        return encoder
+
+    def pair(self, reference: Spectrum, query: Spectrum) -> Tuple[float, float]:
+        """Calculate the MS2DeepScoreMonteCarlo similaritiy between a reference
+        and a query spectrum.
+
+        Parameters
+        ----------
+        reference:
+            Reference spectrum.
+        query:
+            Query spectrum.
+
+        Returns
+        -------
+        ms2ds_similarity
+            MS2DeepScore similarity score.
+        """
+        binned_reference = self.model.spectrum_binner.transform([reference])[0]
+        binned_query = self.model.spectrum_binner.transform([query])[0]
+        reference_vectors = self.get_embedding_ensemble(binned_reference)
+        query_vectors = self.get_embedding_ensemble(binned_query)
+        scores_ensemble = cosine_similarity_matrix(reference_vectors, query_vectors)
+
+        return scores_ensemble.mean(), scores_ensemble.std()
+
+    def matrix(self, references: List[Spectrum], queries: List[Spectrum],
+               is_symmetric: bool = False) -> np.ndarray:
+        """Calculate the MS2DeepScoreMonteCarlo similarities between all references and queries.
+
+        Parameters
+        ----------
+        references:
+            Reference spectrum.
+        queries:
+            Query spectrum.
+        is_symmetric:
+            Set to True if references == queries to speed up calculation about 2x.
+            Uses the fact that in this case score[i, j] = score[j, i]. Default is False.
+
+        Returns
+        -------
+        ms2ds_similarity
```

Comment:
Returns matrix of mean similarity scores and matrix of std's

Reply:
Added this to the documentation (and now also user can chose between median and mean). I recently thought that median would be better, because it is more robust against outliers, so I made median the default.